### ISSUE TITLE
annoyance, Serial slow to start after upload

### ISSUE THREAD

**Issue by tablatronix** [gepd/Deviot#245]
I cannot where I previously wrote this , I am creating a new issue.

It seems the serial console takes 2-3 seconds to start after upload because I miss any serial logging during that time, I have to add a delay to my sketches of at least 2 seconds or I miss output when it resets.

This seems to have changed at some point as it was not always an issue.

Any optimizations here would be worth it I think.

**Comment by gepd** [comment]
Do you keep having this issue?

What S.O were you using?

**Comment by tablatronix** [comment]
Its both osx and win

**Comment by gepd** [comment]
and what board are you using? 
I'm trying with an esp32 and I'm not getting that lag.

Does it happen you with any sketch?

**Comment by tablatronix** [comment]
Yup I have to use a 2000 delay in setup to see the first serial print etc

**Comment by tablatronix** [comment]
Ill make a video

**Comment by tablatronix** [comment]
https://youtu.be/lIDykeUlWrM

**Comment by tablatronix** [comment]
I noticed there is some kinf percentage flashing at the bottom right footer, goes real quick when restarting serial, is it reloading the buffer or something ?

**Comment by gepd** [comment]
I see I'll start to debug what can be causing that delay, I'm not sure if it's the same in my machine, but I'll test.

There is no buffer in the Deviot side. I think it could be the indexing feature from ST

**Comment by tablatronix** [comment]
Hmm Ill disable all plugins and check again
